.github/workflows/pr-build.yml

```diff
@@ -20,8 +20,14 @@ jobs:
         with:
           java-version: 17
       - uses: gradle/wrapper-validation-action@v1
+      - name: Build with Gradle with Integration tests
+        if: ${{ matrix.os == 'ubuntu-latest' }}
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: build integrationTests --stacktrace -PenableCoverage=true -PlocalDocker=true
       - name: Build with Gradle
         uses: gradle/gradle-build-action@v2
+        if: ${{ matrix.os != 'ubuntu-latest' }}
```

Comment:
Why are macos and windows tests executed with different gradle inputs than an ubuntu runner?

Reply:
because they don't support docker.

We were skipping the tests for macos and windows silently because of this:
https://github.com/aws-observability/aws-otel-java-instrumentation/blob/main/smoke-tests/runner/src/test/java/io/awsobservability/instrumentation/smoketests/runner/LogInjectionTest.java#L32
```
@Testcontainers(disabledWithoutDocker = true)
class LogInjectionTest {
```

Se we are not running smoke tests in windows and macos anyways. Just building and running some unit tets.

Reply:
Noted, there are ways we could run docker on MacOS runners. See [here](https://github.com/actions/runner-images/issues/6216). This could possibly be a follow on item if we feel the need to expand test coverage to other operating systems.